src/YamlValidator/ValidatorResults.cs

```diff
@@ -10,6 +12,63 @@ public class ValidatorResults
     public ValidatorResults(bool schemaValid, IReadOnlyList<ValidatorError> traversalResults)
     {
         SchemaValid = schemaValid;
-        TraversalResults = traversalResults;
+        TraversalResults = filterErrors(traversalResults);
+    }
+
+    //  This will filter out the false positives that are not relevant to the error output, when the validation is false
+    private ReadOnlyCollection<ValidatorError> filterErrors(IReadOnlyList<ValidatorError> traversalResults)
```

Comment:
nit: Methods usually start with capital letter

Reply:
Did this out of habit ☹️ . Fixed now